zebra-chain/src/chain_sync_status.rs

```diff
@@ -0,0 +1,32 @@
+//! Defines method signatures for checking if the synchronizer is likely close to the network chain tip.
+
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+
+/// An interface for checking if the synchronization is likely close to the network chain tip.
+pub trait ChainSyncStatus {
+    /// Check if the synchronization is likely close to the network chain tip.
+    fn is_close_to_tip(&self) -> bool;
+}
+
+/// A mock [`ChainSyncStatus`] implementation that allows setting the status externally.
+#[derive(Clone, Default)]
+pub struct MockSyncStatus {
+    is_close_to_tip: Arc<AtomicBool>,
+}
+
+impl MockSyncStatus {
+    /// Sets mock sync status determining the return value of `is_close_to_tip()`
+    pub fn set_is_close_to_tip(&mut self, is_close_to_tip: bool) {
+        self.is_close_to_tip
+            .store(is_close_to_tip, Ordering::Release);
+    }
+}
+
+impl ChainSyncStatus for MockSyncStatus {
+    fn is_close_to_tip(&self) -> bool {
+        self.is_close_to_tip.load(Ordering::Acquire)
```

Comment:
Why did you choose these memory orderings?

Memory orderings are hard to reason about, and the behaviour can vary by architecture, so we usually use the strongest ordering.

If they don't matter because this is a test-only type, please mark the type as `#[cfg(any(test, feature = "proptest-impl"))]`, use `Relaxed`, and document that.

Reply:
> Why did you choose these memory orderings?

I thought it would suffice to retain the order needed but now I feel slightly unsure, and I'd rather wait the extra nanosecond than risk a mistake there, so I replaced it with `SeqCst` in [`6d256c0`](https://github.com/ZcashFoundation/zebra/pull/5769/commits/6d256c01103098bd2388d57e830ed9fb481c25d5), which is what the watch channels are using. `Relaxed` doesn't offer the constraint needed here, though that might not matter for the architectures it's being tested on.

> If they don't matter because this is a test-only type, please mark the type as #[cfg(any(test, feature = "proptest-impl"))], use Relaxed, and document that.

I think this is a really good idea, but I'd like to mark it as `#[cfg(test)]` in another PR alongside other test-only code if possible. I just created issue #5775 for that.

Reply:
Most of the code in issue #5775 is already in modules or crates that are test-only, apart from `NoChainTip`, which I fixed in PR #5777.

Reply:
Thank you!!